plugins/GSdx/GSDevice.cpp

```diff
@@ -45,7 +45,7 @@ GSDevice::GSDevice()
 
 GSDevice::~GSDevice()
 {
-	for(auto &t : m_pool) delete t;
+	for(auto t : m_pool) delete t;
```

Comment:
You can just use `GSTexture*` instead of `auto`. This type is not that hard to read/type, to be worth replacing with `auto`, especially since the other issues.